deployments/Dockerfiles/tendermint/tendermint.py

```diff
@@ -194,22 +197,24 @@ def _start_monitoring_thread(self) -> None:
 
     def _stop_tm_process(self) -> None:
         """Stop a Tendermint node process."""
-        if self._process is None:
+
+        process = self._process
+        if process is None:
             return
 
-        if platform.system() == "Windows":
-            os.kill(self._process.pid, signal.CTRL_C_EVENT)  # type: ignore  # pylint: disable=no-member
-            try:
-                self._process.wait(timeout=5)
-            except subprocess.TimeoutExpired:  # nosec
-                os.kill(self._process.pid, signal.CTRL_BREAK_EVENT)  # type: ignore  # pylint: disable=no-member
-        else:
-            self._process.send_signal(signal.SIGTERM)
-            self._process.wait(timeout=5)
-            poll = self._process.poll()
-            if poll is None:  # pragma: nocover
-                self._process.terminate()
-                self._process.wait(3)
+        process.poll()
+        if process.returncode is not None:
+            return
```

Comment:
Why polling before even calling a function to kill the subprocess?

Reply:
good practice, this function could be called more than once (not in current setup though I believe), and it makes it facilitates readability